docs/DeveloperGuide.md

```diff
@@ -1,38 +1,254 @@
+expenses, help them with budgeting and also be help them to visualise what they have spent.
 
-## Design & implementation
+Overspending and the lack of savings is a significant problem found in most fresh graduates. We want to create tools
+necessary for them so that they can realize their actual spending and achieve financial freedom in the future.
+
+## Setting up the project
+
+## Design
 
 {Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
 
+### Main Component
+
+![MainSeq.png](uml/MainSequence.png)
+
+In the main (Duke) class, the main method first calls startApplication() which initalizes the Log Manager for logging (
+disabled if in production stage) and Ui object for printing output messages. After initialized, it will then attempt to
+import the user's data from `data.json`. If there is no file, it will initalize a blank data. If importing fails, the
+application will print an error message and exit with code 1. Once done, it will greet the user with a message.
+
+The application then runs an infinite loop to take in and execute commands, until when the inputs `exit` to exit the
+application. After which, it will output a bye message and exits the application safely.
+
+### Command Component
+
+![Command.png](uml/Command.png)
+
+The command consists of three components: Enum CommandEnum, Abstract class Command and Class CommandParser. Each of them
+plays a role in retrieving the commands input by the user and redirecting to the correct components.
+
+#### CommandEnum
+
+The CommandEnum Enum contains all the commands that are available in the application.
+These are the only commands that can be input in the command line.
+
+#### Command
+
+The abstract class Command extends to several commands (eg `BudgetCommand`, `DepositCommand`, `ExpenseCommand`) and contains abstract methods execute() and isExit().
+The execute() method is executed from main, where it will go to the action class (eg `BudgetAction`) and run the requested action such as `add`.
+After execute() from main, isExit() controls the termination of BudgetBuddy by returning a boolean.
+
+#### CommandParser
+
+![CommandParserSequence.png](uml/CommandParserSequence.png)
+
+The `CommandParser` class takes in an input from the `Duke` class as seen in the sequence diagram above. It then parses
+the information
+to create a command of a specific type (eg `BudgetCommand`, `DepositCommand` , explained in the class diagram above)
+based
+the input.
 
 ## Product scope
-### Target user profile
 
-{Describe the target user profile}
+### Item Component
+
+The main 3 classes of Duke are the `budget` , `expense` and `deposit` class. Users are able to add, store and visualise
+the date relate to each
+of these classes. Each of these classes are modelled as an `Item`.
+
+![ItemClasses.png](uml/ItemClasses.png)
```

Comment:
Shouldn't you give an access modifier?

<img width="935" alt="image" src="https://user-images.githubusercontent.com/81511355/228762894-77e45fbd-7e68-4300-b34e-e1f79922bf04.png">